virtwho/virt/xen/xen.py

```diff
@@ -45,6 +45,10 @@ def __init__(self, section_name, wrapper, *args, **kwargs):
         self.add_key('server', validation_method=self._validate_server, required=True)
         self.add_key('username', validation_method=self._validate_username, required=True)
         self.add_key('password', validation_method=self._validate_unencrypted_password, required=True)
+        # strange issue with XEN API and the use of this environment variable
```

Comment:
Interesting. I wonder, how is modified 'no_proxy' used for connecting Xen server? I can see that `server`, `username` and `password` are used, but I cannot see that `no_proxy` is used or anything else from `self.config`.

Is there any documentation or explanation, why should `no_proxy` should start with ','?

I would implement some method for validation `no_proxy` and I would return some info/debug about changed `no_proxy`. Something that we use for in `_validate_server()`. Thoughts?

Reply:
It appears to be read as an environment variable. When run as a service that means the values in /etc/sysconfig/virt-who.

We do not set it on the connection to Xen via the XenApi. It must just get used quitely. 
The value set in *.conf files is rhsm_no_proxy and only affects the connection for subscription-manager to server communication.